apps/system/js/statusbar.js

```diff
@@ -431,6 +440,12 @@
         case 'cardviewshown':
         case 'cardviewclosed':
           return 'cardview';
+        case 'appwillopen':
+        case 'stackchanged':
+          return 'appopening';
+        case 'appwillclose':
+        case 'stackchanged':
```

Comment:
Looks like the app_transition_controller might also be dispatching an `appclose` (without the _d_) event. It's messy.
But I think it make sense to check and to use app_transition_controller events, they should be reliable since it's a state machine.